internal/controller/factory/statefulset.go

```diff
@@ -221,3 +194,65 @@ func generateEtcdCommand(cluster *etcdaenixiov1alpha1.EtcdCluster) []string {
 
 	return command
 }
+
+func getStartupProbe(probe *corev1.Probe) *corev1.Probe {
+	defaultProbe := &corev1.Probe{
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path: "/readyz?serializable=false",
+				Port: intstr.FromInt32(2379),
+			},
+		},
+		InitialDelaySeconds: 1,
+		PeriodSeconds:       5,
+	}
+	return mergeWithDefaultProbe(probe, defaultProbe)
+}
+
+func getReadinessProbe(probe *corev1.Probe) *corev1.Probe {
+	defaultProbe := &corev1.Probe{
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path: "/readyz",
+				Port: intstr.FromInt32(2379),
+			},
+		},
+		InitialDelaySeconds: 5,
+		PeriodSeconds:       5,
+	}
+	return mergeWithDefaultProbe(probe, defaultProbe)
+}
+
+func getLivenessProbe(probe *corev1.Probe) *corev1.Probe {
+	defaultProbe := &corev1.Probe{
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path: "/livez",
+				Port: intstr.FromInt32(2379),
+			},
+		},
+		InitialDelaySeconds: 5,
+		PeriodSeconds:       5,
+	}
+	return mergeWithDefaultProbe(probe, defaultProbe)
+}
+
+func mergeWithDefaultProbe(probe *corev1.Probe, defaultProbe *corev1.Probe) *corev1.Probe {
```

Comment:
I think usage of pointers is bad here as `defaultProbe` is altered and then returned. Function is not pure

Reply:
Good notice, fixed